iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/extensions/AbstractReferenceTerminalsTest.java

```diff
@@ -196,6 +196,55 @@ public void testWrongNetwork() {
         assertEquals("Terminal given is not in the right Network (sim1 instead of fourSubstations)", ex2.getMessage());
     }
 
+    @Test
+    public void testWithSubnetwork() {
+        Network merged = Network.merge(network, EurostagTutorialExample1Factory.create());
```

Comment:
Could you add unit tests using the listener?

Reply:
added a specific test with merge and detach

Reply:
Thanks!